### DISCUSSION TITLE
twinBASIC pre-order information!

### DISCUSSION THREAD

**Discussion by WaynePhillipsEA** [twinbasic/twinbasic#281] [General]
Later this week we'll start accepting pre-orders for twinBASIC Professional Edition and Ultimate Edition.

For pricing and more information, please visit:
https://twinbasic.com/preorder.html

If you have any questions, please ask them here.  Thanks as always go to you all for your help in testing twinBASIC so far and for your awesome feedback. 

### EDIT: The pre-order offer is now live:
https://www.twinbasic.com/preorder.html

**Comment by Kr00l** [comment]
Will it be possible to have, beside a monthly subscription, a yearly subscription?
Some people prefer a yearly charge instead of monthly to reduce the amount of transactions..
Also a yearly fee could come with a tiny discount. (Like on a car insurance for example)

**Reply by DavidHollman** [reply]
It may even be more true for people in large corporations that a single annual transaction would be way easier than numerous recurring ones. Paying for external things sometimes is hugely bureaucratic in some companies. The cost probably won't be an issue but the overhead of dealing with it could discourage people from doing the right thing...

**Reply by WaynePhillipsEA** [reply]
Both very valid points that we are considering, thanks.   We'll get back to you on it tomorrow.

**Reply by WaynePhillipsEA** [reply]
Thanks for the feedback guys.  After consideration, we've decided to offer the annual option at a 10% discount rate.  For the annual plans, payment will also be possible via bank transfer without auto-renew.

**Comment by WaynePhillipsEA** [comment]
The pre-order offer is now live, folks:
https://www.twinbasic.com/preorder.html

For orders via bank transfer, we will contact you by email within 24-48 hours in order to provide a pro-forma invoice.

**Comment by dmrvb** [comment]
Can you tell us a bit more about how this works please?
I don't really grasp the concept of a pre-order on a subscription.
I don't want to start paying a subscription now, not till you have something that will compile all of my code and produce an .EXE that is at least a bit better than what VB can do. The cross platform stuff is still a year away, maybe more, that's a big risk.
Are you really just asking for a commitment to subscribe when you do have it all working?.
How are you going to manage the licences? will you hand out code on a trust basis or will it all be tied in to a server?

**Reply by WaynePhillipsEA** [reply]
Sure.   The pre-order offer is simply a chance to lock in now at a significantly discounted monthly rate, before the work is complete.  That's the long term benefit to you - you'll continue to get the discounted monthly rate indefinitely (whilst new sign-ups will pay the full price once the pre-order offer finishes). 

For us, this allows us to better manage our resources right now, potentially bringing forward work by being able to outsource some of it, depending on the pre-order uptake of course.

I appreciate that it requires some level of trust in ordering something before it is actually finished. 
Hopefully you'll have seen enough of the quality of our work on twinBASIC to trust that we know what we are doing and will get twinBASIC to where we all want it to be.

It is planned that licences will require a one-off online activation.

If you have any further questions, please do let me know.

**Reply by dmrvb** [reply]
That all sounds good to me.
This is a big project so some outsourcing might be good, but taking on an assistant (possibly self employed at first) might be better in the longer term.
I really don't know how big the potential audience is for TwinBasic, but I suspect that so far only a tiny fraction are aware of what you are doing, I only found TwinBasic by a very convenient co-incidence.

**Reply by WaynePhillipsEA** [reply]
I agree.  I think (hope) we will start to gain more traction once we have some GUI support, and 64-bit support is in place.

**Reply by dmrvb** [reply]
The GUI is pretty much essential to get the project really going but I think you will still need some well thought out publicity to reach a significant audience. The potential users are probably very diverse so it will need a bit of clever thinking to communicate, you will know much more about this than me, but maybe some others here have some ideas too.

**Comment by WaynePhillipsEA** [comment]
Epic fail... 

Any orders placed yesterday with the 'Yearly (bank transfer)' option didn't get logged.   The problem has been fixed - please accept my apologies.  Please re-submit the form and we will get back to you with the pro-forma invoice.

(all PayPal orders came through fine, so no need to re-submit them, thanks).

**Comment by Kr00l** [comment]
Would it be possible to subscribe to pro edition (cross-platform not an requirement now) and possibly upgrade to ultimate edition at a later stage (or not who knows) but to keep the 50% pre-order discount ?

**Reply by WaynePhillipsEA** [reply]
Good question.  That's a tough one, as the early-bird discount was designed as an incentive to help fund some of that cross-platform work.  After some consideration, we've decided to allow moving from the Pro-edition to the Ultimate-edition, but there will be a reduced discount of 40%, so the cost in GBP would then be £36/mo after the change.  If you're on a yearly plan, we'll honour the same 40% deal on a pro-rata refund basis.

**Reply by WaynePhillipsEA** [reply]
We'll update the preorder webpage tomorrow with this extra information.

**Comment by dmrvb** [comment]
I very much feel the same as Kr001.
My main cross platform interest would be iPad and iPhone and I don't think these are even on your list yet, but I would love to retain my pre-order discount if you do include these at some time in the future.

**Reply by WaynePhillipsEA** [reply]
Once the bulk of the cross-platform work is complete, adding new platforms will become significantly easier.

**Comment by mwolfe02** [comment]
Does the Standard edition allow for commercial use?  Are the licenses royalty-free?

Whatever the answers are, I would suggest clarifying those two points on the pre-order page (maybe in the feature comparison table).

**Reply by WaynePhillipsEA** [reply]
Yes and yes.  Now clarified on the pre-order page.  Thanks

**Comment by VB6-Programming** [comment]
Can you clarify the status of the GUI Forms designer. You seem to be saying it will be available in the first release (in 2-3 months) but also that full backwards compatibility with VB6 will be 6-12 months.
So will we be able to import VB6 forms and code in the first release?

**Reply by WaynePhillipsEA** [reply]
The first available forms will be ones that are designed for cross-platform support (some more detail in #270).  These initial forms will not offer full backwards-compatibility.   Migrating/importing simple VB6 forms to the cross-platform forms should be possible at that stage.

Fully compatible VB6 forms will be worked on after the initial release, with full backwards compatibility available within the 12 months.

**Reply by dmrvb** [reply]
I think I don't agree with this approach. (smiley)
100% VB6 compatibility should be Top priority because that is what will get TB really going. This must include compatibility with all existing VB6 forms, including the complicated ones. XOJO and others have been around for years and have not really taken off because they are not compatible (and other reasons).
I do have a vested interest as I have a lot of VB6 code, but I suggest that you can not really get any income from TB Pro until it is 100% compatible, so by mixing the Pro with the cross platform (12 months+ away) you are delaying the point at which the money starts flowing in.

**Reply by WaynePhillipsEA** [reply]
Thanks for your thoughts.  In an ideal world I agree, but there are the practical things to consider.   

Creating a fully compatible VB6 forms engine requires significant investment of time due to the precision needed to make every form and control property and event match the existing VB6 behaviours exactly.  Hence the much longer timeframe for this part of the development. 

On the other hand, new, cross-platform based forms can be created much quicker as we are not having to conform to an existing standard.  The plan is to get the talented community here involved in creating the cross-platform forms/controls, which also lessens the timeframe.

There is plenty of overlap in the initial work of the cross-platform forms that will also be used for the VB6-compatible forms - in particular, the form designer.  So in all, the development cost to start on the cross-platform forms is actually small and wouldn't massively affect the timeframe for the VB6-compatible forms anyway.

**Reply by dmrvb** [reply]
This is a big and complicated decision, I'm glad its you and not me who has to make it. 100% compatibility with VB, including the GUI, would be a HUGE bonus and massively increase the prospect of success for TB. Is it achievable?  I did a lot of detailed stuff with the Treeview control a while ago and eventually gave up, there is just too much difficult behaviour in the Microsoft controls.
Will your cross platform controls still use the native Windows controls (to give the correct "look and feel") or do you intend to make a whole set of brand new controls? New controls will be a big job but maybe not such a disadvantage as they would have been a few years ago.
Is it actually possible to use the current OCX controls from a 64 bit compilation?

**Comment by eantar** [comment]
Hey Wayne, this stuff is incredible.  But, I'm primarily an Access programmer. It's taking me a while to get around VS Code. I was wondering if you guys might put out a video exclusively focused on getting Access programmers used to just working in the VBA IDE working in VS Code with the twinBasic extension. Thanks.

**Reply by WaynePhillipsEA** [reply]
I've opened a separate discussion to tackle this: #297, please let me know your thoughts in that discussion.

Videos do take quite some time to put together, so we'll need as much information from you as possible in order to make sure we're targeting the right things.
